### ISSUE TITLE
IPython disk I/O error

### ISSUE THREAD

**Issue by chebee7i** [ipython/ipython#8520]
I'm having some trouble building Sphinx docs, and I'm not quite sure what is going on, but it is definitely related to the IPython directive. The problem seems to have started when moving away from Sphinx 1.2 and onto Sphinx 1.3. After that, the IPython directive was no longer able to receive output. That is, in https://github.com/ipython/ipython/blob/master/IPython/sphinxext/ipython_directive.py#L463, the variable `processed_output` was always ''. This is obviously going to cause problems. 

I created a new virtual environment and tried see if I could replicate this. Instead of getting the same issue, I received:

```
Error in sys.exitfunc:
Traceback (most recent call last):
  File "/usr/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/home/me/.virtualenvs/ipytest/local/lib/python2.7/site-packages/IPython/core/interactiveshell.py", line 3363, in atexit_operations
    self.history_manager.end_session()
  File "/home/me/.virtualenvs/ipytest/local/lib/python2.7/site-packages/IPython/core/history.py", line 540, in end_session
    len(self.input_hist_parsed)-1, self.session_number))
sqlite3.OperationalError: disk I/O error
```

Deleting my history file and the sphinx build directory did not fix the issue. So I'm having a bit of trouble pinning this down and I thought I'd see if anyone had any ideas first. This seems to be two separate issues, but perhaps they are related.

**Comment by chebee7i** [comment]
Additional information:  The `OperationalError` is not present in the development version of IPython but it does exist in 3.1.0 (using Sphinx 1.3 for both scenarios).  So after upgrading to the dev version, the issue with `processed_output` is visible again.

**Comment by chebee7i** [comment]
Ok, I've managed to avoid the issue. It turns out I had some old test files from when I was doing some work on the IPython directive. If you have a block in an rst file such as:
```
.. ipython::

   In [2]: x = (

   0,

   1)
```

which is only possible through a pasting via Ctrl-V (in an old version of IPython)...then it would cause weird errors like I described above in sections of code that are unrelated to it. The exact error would depend on the version of Sphinx as well. 

Anyway, the above is no longer valid input (even when using Ctrl-V or %paste), so I think we can just ignore this issue..unless it comes up again.
